src/allmydata/test/cli/wormholetesting.py

```diff
@@ -276,8 +281,8 @@ def when_received(self) -> Deferred[WormholeMessage]:
             )
         d = self._view.wormhole_by_code(self._code, exclude=self)
 
-        def got_wormhole(wormhole):
-            msg = wormhole._payload.get()
+        def got_wormhole(wormhole: _MemoryWormhole) -> Deferred[_MemoryWormhole]:
+            msg: Deferred[_MemoryWormhole] = wormhole._payload.get()
```

Comment:
`wormhole._payload` had a `WormholeMessage` put in it on line 269... I hope this doesn't type check?

Reply:
Looking at this again, it seems to type check because in Twisted <21.2.0 `DeferredQueue` is not a "Generic" type (ie, it is not polymorphic).  Thus `wormhole._payload` has type `DeferredQueue` and `wormhole._payload.get` has type `Any`.  So, this type checks just fine because mypy can't tell any better.

However, it is still incorrect.  The correct return annotation for `got_wormhole` is `Deferred[WormholeMessage]` and the correct annotation for `msg` is the same.

Reply:
Okay thank you I will include it in my next fix.